### PR TITLE
Product crud ajax

### PR DESCRIPTION
**Pull request by linktotalha**

### CODE REVIEW

File: app/Http/Controllers/ProductController.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            'name'=>$req->name,
+            'price'=>$req->price,
+            'desc'=>$req->desc,
+            'user_id'=>Auth::id()
+        ]);
+
+        foreach($req->image as $image){
+            $product->images()->create([
+                'product_id'=>$product->id,
+                'image'=>$image
```
Comment (yasirmehmood123) [inline app/Http/Controllers/ProductController.php]: what is saved in this $image?

File: database/migrations/2023_03_09_065552_create_products_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->float('price');
+            $table->text('desc');
+            $table->unsignedBigInteger('user_id');
```
Comment (yasirmehmood123) [inline database/migrations/2023_03_09_065552_create_products_table.php:21]: unsigned id should be right after id

File: database/migrations/2023_03_09_071629_create_mediaables_table.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class CreateMediaablesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('mediaables', function (Blueprint $table) {
```
Comment (yasirmehmood123) [inline database/migrations/2023_03_09_071629_create_mediaables_table.php:16]: what is the reason of creating polymorphic table

File: resources/views/products/create.blade.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                    <span class="descErr text-danger"></span>
+                </div>
+                <button type="submit" id="submitBtn" class="btn btn-primary">Submit</button>
+            </form>
+        </div>
+    </div>
+@endsection
+
+@section('scripts')
+    <script src="https://code.jquery.com/jquery-3.6.3.min.js"
```
Comment (yasirmehmood123) [inline resources/views/products/create.blade.php:122]: use NPM for this

File: routes/web.php [hunk 5]
```diff
@@ -18,16 +19,21 @@
...
     Route::post('categories/create',[CategoryController::class,'postCreate']);
     Route::get('category-list',[CategoryController::class,'list']);
     Route::get('delete-category',[CategoryController::class,'delete']);
     Route::get('edit-category',[CategoryController::class,'edit']);
     Route::post('edit-category',[CategoryController::class,'postEdit']);
 
-    Route::get('/dashboard', function () {
-        return view('dashboard');
-    });
+    // product routes
```
Comment (yasirmehmood123) [inline routes/web.php:34]: use resource route

File: app/Http/Controllers/ProductController.php [hunk 6]
```diff
@@ -26,20 +31,70 @@ public function postCreate(Request $req){
...
+            $product->categories()->create([
+                'category_id'=>$category
+            ]);
+        }
+        
+
+        // multiple images
         foreach($req->image as $image){
+            $imageName = Str::random(4).time().'.'.$image->extension();
+            $image->move(public_path('images'), $imageName);
```
Comment (yasirmehmood123) [inline app/Http/Controllers/ProductController.php:53]: in the instruction that was specified that •	Use intervention package

File: database/migrations/2023_03_09_105956_create_product_categories_table.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+use Illuminate\Support\Facades\Schema;
+
+class CreateProductCategoriesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
```
Comment (yasirmehmood123) [inline database/migrations/2023_03_09_105956_create_product_categories_table.php:14]: use pivot

**Review by yasirmehmood123 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/ProductController.php (hunk 1 above)
Comment (linktotalha) [reply app/Http/Controllers/ProductController.php]: changes ki hain code push krna h abhi iska

File: database/migrations/2023_03_09_065552_create_products_table.php (hunk 2 above)
Comment (linktotalha) [reply database/migrations/2023_03_09_065552_create_products_table.php:21]: ok

File: database/migrations/2023_03_09_071629_create_mediaables_table.php (hunk 3 above)
Comment (linktotalha) [reply database/migrations/2023_03_09_071629_create_mediaables_table.php:16]: ye delete krdiye hain phely hi

**Pull request closed without merging**